pkg/command/root.go

```diff
@@ -300,6 +300,11 @@ func setupTargetPlugins() error {
 		if targetCmd, exists := mapTargetToCmd[plugins[i].Target]; exists {
 			cmd := cli.GetUnmappedCmdForPlugin(&plugins[i])
 			if cmd != nil {
+				if plugins[i].Target == configtypes.TargetK8s {
+					// Add a deprecation message which will be printed when the user invokes
+					// any of the plugins commands using the form `tanzu kubernetes <plugin> ...`
+					cmd.Deprecated = fmt.Sprintf(`when called with the "kubernetes" prefix. Please invoke %q directly without the "kubernetes" prefix.`, cmd.Name())
```

Comment:
Because cobra is opinionated about how it wants to format the deprecation message, this ends up reading a little awkward. I would have preferred that the messages says that invoking a command with k8s prefix is deprecated, not that the command itself is. 

I don't think we can do much about it. But one option I am considering is not even showing this message at all...

Reply:
Yeah, this is not ideal and it annoys me that Cobra enforces a prefix instead of letting us override completely.  Something we should improve in Cobra.

Until then we could could remove this deprecation and simply hide the command.
To be honest, I had started with this deprecation before figuring out how to get a deprecation message on all sub-commands (which Cobra does not support, but that we can pull off since we only have a single command for a plugin).

What do you prefer?

Reply:
> (which Cobra does not support, but that we can pull off since we only have a single command for a plugin).
I forgot to also mention, the prefixing + custom string, when, coupled with the fact that we only reference the command name for the plugin (and not the actual subcommand invoked), is what makes the message feel awkward. 

What are the downsides for just hiding instead of showing the deprecation message? I figure for folks navigating from top level "tanzu", they wouldn't have seen the `kubernetes` command anyway, so it is no biggie. It appears what we lose is the ability to flag existing/established use of this form (e.g in scripts) as deprecated.

Reply:
> It appears what we lose is the ability to flag existing/established use of this form (e.g in scripts) as deprecated.

I agree that is the disadvantage.  Originally I was just going to hide the `tanzu kubernetes` command and that was it, but the requirement asked for a deprecation message.  We could revisit the requirement 😄

Reply:
So there was a couple of things in the comment and I may not have been clear. I was thinking of keeping the deprecation notice for `tanzu kubernetes` if there is a way to make it still work as before (show child commands under it)
I am debating on the merits of just letting `tanzu kubernetes xxx...` run without a deprecation message because the message looks awkward atm. We lose a bit not having the deprecation message, but we can always come back and add it later as well.

Reply:
Let's start with that then.  Only hide the `kubernetes` target and keep its sub commands like before.